### DISCUSSION TITLE
How can I avoid a `link` for categories generated by tag

### DISCUSSION THREAD

**Discussion by IanVS** [PaloAltoNetworks/docusaurus-openapi-docs#434] [Q&A] [answered]
I do not want to have the tag categories as actual pages, just headings which collapse and expand when clicked.  I can do this with other sidebar items by not giving the category a `link`.  I thought this might be the default behavior when `categoryLinkSource` is not set, but it seems that's the same as setting it to `tag`.  Is that right?  Is there a way to accomplish what I'm looking for, or do we need a new option like `"none"` or something?

**Comment by sserrata** [comment, marked as answer]
Hi @IanVS, I think the default behavior should be to disable the category link when `categoryLinkSource` is not set or is `undefined`.

**Reply by IanVS** [reply]
I agree, though that might be a disruptive change for those relying on the current behavior.

**Reply by sserrata** [reply]
Agreed. For that reason, it's probably best to introduce the change in the upcoming v2.0.0 release. For now, the work around, although not ideal, is manually editing the generated `sidebar.js`.

**Reply by IanVS** [reply]
That won't work for me, since I gitignore the generated files and re-generate them during deployment.  Would you be open to a `none` option for now, until it can be re-worked in 2.0?

**Reply by sserrata** [reply]
Sure, I think we can do that. 👍🏽

**Reply by IanVS** [reply]
If you'd like to point me in the right direction, I'm happy to put together a PR if that's helpful.

**Reply by IanVS** [reply]
@sserrata is this something you'd be willing to accept a PR for?

**Reply by sserrata** [reply]
Sure, a PR would be very welcome

**Reply by IanVS** [reply]
Since you're making a breaking change now with 2.0, should I submit another PR to make the default behavior to not create a link page, as we discussed above?

**Reply by sserrata** [reply]
Hi @IanVS, this is absolutely a good time for this change and I agree with the change.

**Reply by IanVS** [reply]
@sserrata I took a closer look, and it seems that the current default is somewhat different from either of the other options `"tag"` or `"info"`.  Do we need a new option for what was previously the default?  Maybe `"auto"`?  I'm not entirely clear on the differences between all three current options.  If you can give some guidance, I'm happy to put together a PR.

**Reply by sserrata** [reply]
Yes, the current default should be to simply set the category link to a generated-index. The "tag" option is similar except that it actually renders a doc/page with the tag description at the top and the generated-index below. The "info" option is a bit of a special case that came up in our product APIs, whereby a single API is broken up into micro API specs. When you want to render all micro APIs in the same sidebar you can use the "info" option to render a doc/page with the OpenAPI info description at the top and the generated-index below.

**Reply by IanVS** [reply]
Thanks @sserrata, so if the new default will be to create just a spindown (as I have on docs.defined.net currently), should the _current_ default still be possible to achieve somehow?  If so, should we give it a new option name? If so, any suggestions?  Perhaps `"auto"` or something?

**Reply by sserrata** [reply]
Great question. The most intuitive option might be something like "generated-index" or "index-only" but it still feels a bit convoluted IMO. I like "auto" too but it may be less obvious than the aforementioned. To recap:

- "tag": Tag description if present + `generated-index`
- "info": Info description if present + `generated-index`
- "auto" or "generated-index": Returns only `generated-index`
- `null` or `undefined`: No category links
